### ISSUE TITLE
Checking lambda raises "undefined method `module_creation_call?`"

### ISSUE THREAD

**Issue by alexgerstein** [troessner/reek#960]
When running reek on a lambda set to a constant, the context builder raises a NoMethodError.

Example Code:

```
`MISSING_TRANSLATION_HANDLER = -> (exception, _locale, _key, _options) { raise exception.to_exception }`
```

Backtrace:

```
gems/reek-4.1.0/lib/reek/ast/sexp_extensions/module.rb:58:in `defines_module?': undefined method `module_creation_call?' for s(:lambda):#<Class:0x007f7fb8fc1248> (NoMethodError)
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/gems/reek-4.1.0/lib/reek/context_builder.rb:106:in `process_casgn'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/gems/reek-4.1.0/lib/reek/context_builder.rb:63:in `build'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/gems/reek-4.1.0/lib/reek/context_builder.rb:30:in `initialize'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/gems/reek-4.1.0/lib/reek/examiner.rb:80:in `new'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/gems/reek-4.1.0/lib/reek/examiner.rb:80:in `run'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/gems/reek-4.1.0/lib/reek/examiner.rb:37:in `initialize'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/gems/reek-4.1.0/lib/reek/cli/command/report_command.rb:24:in `new'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/gems/reek-4.1.0/lib/reek/cli/command/report_command.rb:24:in `block in populate_reporter_with_smells'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/gems/reek-4.1.0/lib/reek/cli/command/report_command.rb:23:in `each'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/gems/reek-4.1.0/lib/reek/cli/command/report_command.rb:23:in `populate_reporter_with_smells'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/gems/reek-4.1.0/lib/reek/cli/command/report_command.rb:15:in `execute'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/gems/reek-4.1.0/lib/reek/cli/application.rb:28:in `execute'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/gems/reek-4.1.0/bin/reek:13:in `<top (required)>'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/bin/reek:23:in `load'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/bin/reek:23:in `<main>'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/bin/ruby_executable_hooks:15:in `eval'
	from /Users/alex.gerstein/.rvm/gems/ruby-2.2.2@zenpayroll/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by troessner** [comment]
Confirmed bug. Easily reproducible:

```
echo "C = -> { }" | reek
```

gives:

```
/Users/timo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/reek-4.1.0/lib/reek/ast/sexp_extensions/module.rb:58:in `defines_module?': undefined method `module_creation_call?' for s(:lambda):#<Class:0x007fa23da598f8> (NoMethodError)
	from /Users/timo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/reek-4.1.0/lib/reek/context_builder.rb:106:in `process_casgn'
	from /Users/timo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/reek-4.1.0/lib/reek/context_builder.rb:63:in `build'
	from /Users/timo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/reek-4.1.0/lib/reek/context_builder.rb:30:in `initialize'
	from /Users/timo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/reek-4.1.0/lib/reek/examiner.rb:80:in `new'
	from /Users/timo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/reek-4.1.0/lib/reek/examiner.rb:80:in `run'
	from /Users/timo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/reek-4.1.0/lib/reek/examiner.rb:37:in `initialize'
	from /Users/timo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/reek-4.1.0/lib/reek/cli/command/report_command.rb:24:in `new'
	from /Users/timo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/reek-4.1.0/lib/reek/cli/command/report_command.rb:24:in `block in populate_reporter_with_smells'
	from /Users/timo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/reek-4.1.0/lib/reek/cli/command/report_command.rb:23:in `each'
	from /Users/timo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/reek-4.1.0/lib/reek/cli/command/report_command.rb:23:in `populate_reporter_with_smells'
	from /Users/timo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/reek-4.1.0/lib/reek/cli/command/report_command.rb:15:in `execute'
	from /Users/timo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/reek-4.1.0/lib/reek/cli/application.rb:28:in `execute'
	from /Users/timo/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/reek-4.1.0/bin/reek:13:in `<top (required)>'
	from /Users/timo/.rbenv/versions/2.3.0/bin/reek:23:in `load'
	from /Users/timo/.rbenv/versions/2.3.0/bin/reek:23:in `<main>'
```

Thanks for the report, we'll look into it!

**Comment by troessner** [comment]
Just released 4.1.1 with the bugfix from #963 - can you please check if this works for @alexgerstein ?
